services-camel/src/test/java/edu/unc/lib/dl/services/camel/routing/MetaServicesRouterTest.java

```diff
@@ -78,6 +84,110 @@ public void testRouteStartContainer() throws Exception {
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(0);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent(CONTAINER_ID + "/datafs",
+                Fcrepo4Repository.Container.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartDepositRecord() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(0);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent(DEPOSIT_ID,
+                Fcrepo4Repository.Container.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartNotAPid() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(0);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("what/is/going/on",
+                Fcrepo4Repository.Container.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartCollections() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(0);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("/content/" + RepositoryPathConstants.CONTENT_ROOT_ID,
+                Fcrepo4Repository.Container.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartBinaryMetadata() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(0);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent(CONTAINER_ID + "/datafs/original_file/fcr:metadata",
+                Fcrepo4Repository.NonRdfSourceDescription.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartOriginalBinary() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(1);
+
+        createContext(META_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent(CONTAINER_ID + "/datafs/original_file",
+                Fcrepo4Repository.Binary.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testRouteStartPremisBinary() throws Exception {
+        getMockEndpoint("mock:direct-vm:index.start").expectedMessageCount(1);
+        getMockEndpoint("mock:direct:process.enhancement").expectedMessageCount(1);
```

Comment:
Should event logs have enhancements run on them?

Reply:
The issue at the moment is that longleaf registration happens in the EnhancementRouter, while the new filtering is in MetaServicesRouter, so all binaries have to reach MetaServicesRouter at the moment. But we don't need enhancements on any binaries except originals currently. I could move the call to longleaf up into the metaservicesrouter... that might be better in some ways, as running enhancements doesn't affect the original binary, so it shouldn't need to register in longleaf again. On the other hand, we might have to add a way to trigger longleaf registration for cases where it fails. I think that's fine though. I'll try to split longleaf out of the enhancement router.